### ISSUE TITLE
Support multi-tenant single database where each team have schema

### ISSUE THREAD

**Issue by rdagumampan** [rdagumampan/yuniql#67]
In a multi-tenant database where schema is used as unit of isolation, multiple users requires separate version control mechanism in objects within their schema. 

For example, in a Data Warehouse where there are three teams each work independently, there must be three version control mechanism also. Each team works to evolve all the db objects in their schema.

**Comment by rdagumampan** [comment]
## Example for Postgres

Deploy postgresql container

```console
docker run -dit --name yuniql-postgresql -e POSTGRES_USER=sa -e POSTGRES_PASSWORD=P@ssw0rd! -e POSTGRES_DB=helloyuniql -p 5432:5432 postgres
```

Setup schema versioning for West Europe database tenant
```console
yuniql run -a -p C:\play\yuniql-multi-tenant\samples\postgresql-multi-tenant-sample -c "Host=localhost;Port=5432;Username=sa;Password=P@ssw0rd!;Database=helloyuniql" --platform postgresql --debug --meta-schema weu -k "TENANT_RESERVED_NAME=weu"
```

List versions
```console
yuniql list -c "Host=localhost;Port=5432;Username=sa;Password=P@ssw0rd!;Database=helloyuniql" --platform postgresql --debug  --meta-schema weu
```

Setup schema versioning for North Europe database tenant
```console
yuniql run -a -p C:\play\yuniql-multi-tenant\samples\postgresql-multi-tenant-sample -c "Host=localhost;Port=5432;Username=sa;Password=P@ssw0rd!;Database=helloyuniql" --platform postgresql --debug --meta-schema neu  -k "TENANT_RESERVED_NAME=neu"
```

List versions
```console
yuniql list -c "Host=localhost;Port=5432;Username=sa;Password=P@ssw0rd!;Database=helloyuniql" --platform postgresql --debug  --meta-schema neu
```

Setup schema versioning for Asia Pacific database tenant
```console
yuniql run -a -p C:\play\yuniql-multi-tenant\samples\postgresql-multi-tenant-sample -c "Host=localhost;Port=5432;Username=sa;Password=P@ssw0rd!;Database=helloyuniql" --platform postgresql --debug --meta-schema apac  -k "TENANT_RESERVED_NAME=apac"
```

List versions
```console
yuniql list -c "Host=localhost;Port=5432;Username=sa;Password=P@ssw0rd!;Database=helloyuniql" --platform postgresql --debug  --meta-schema apac
```

Cleanup West Europe database tenant

```console
yuniql erase --force -p C:\play\yuniql-multi-tenant\samples\postgresql-multi-tenant-sample -c "Host=localhost;Port=5432;Username=sa;Password=P@ssw0rd!;Database=helloyuniql" --platform postgresql --debug --meta-schema weu -k "TENANT_RESERVED_NAME=weu"
```

Cleanup North Europe database tenant

```console
yuniql erase --force -p C:\play\yuniql-multi-tenant\samples\postgresql-multi-tenant-sample -c "Host=localhost;Port=5432;Username=sa;Password=P@ssw0rd!;Database=helloyuniql" --platform postgresql --debug --meta-schema neu  -k "TENANT_RESERVED_NAME=neu"
```

Cleanup Asia Pacific database tenant

```console
yuniql erase --force -p C:\play\yuniql-multi-tenant\samples\postgresql-multi-tenant-sample -c "Host=localhost;Port=5432;Username=sa;Password=P@ssw0rd!;Database=helloyuniql" --platform postgresql --debug --meta-schema apac  -k "TENANT_RESERVED_NAME=apac"
```

Destroy West Europe database tenant

```console
yuniql destroy --force -c "Host=localhost;Port=5432;Username=sa;Password=P@ssw0rd!;Database=helloyuniql" --platform postgresql --debug
```

**Comment by rdagumampan** [comment]
Done. This feature will be released as part of v1.2. Stay tuned!
Sample for SqlServer

```
docker run -e "ACCEPT_EULA=Y" -e "MSSQL_SA_PASSWORD=P@ssw0rd!" -p 1400:1433 -d mcr.microsoft.com/mssql/server:2017-latest

yuniql run -a -p "c:\play\yuniql-multi-tenant\samples\sqlserver-multi-tenant-sample" -c "Server=localhost,1400;Database=helloyuniql;User Id=SA;Password=P@ssw0rd!" --debug  --meta-schema weu  -k   "TENANT_RESERVED_NAME=weu"

yuniql run -a -p "c:\play\yuniql-multi-tenant\samples\sqlserver-multi-tenant-sample" -c "Server=localhost,1400;Database=helloyuniql;User Id=SA;Password=P@ssw0rd!" --debug  --meta-schema neu  -k "TENANT_RESERVED_NAME=neu"

yuniql run -a -p "c:\play\yuniql-multi-tenant\samples\sqlserver-multi-tenant-sample" -c "Server=localhost,1400;Database=helloyuniql;User Id=SA;Password=P@ssw0rd!" --debug  --meta-schema apac  -k "TENANT_RESERVED_NAME=apac"

yuniql list -c "Server=localhost,1400;Database=helloyuniql;User Id=SA;Password=P@ssw0rd!" --debug --platform sqlserver

yuniql erase -p "c:\play\yuniql-multi-tenant\samples\sqlserver-multi-tenant-sample"  -f -c "Server=localhost,1400;Database=helloyuniql;User Id=SA;Password=P@ssw0rd!" --debug --platform sqlserver  --meta-schema weu  -k "TENANT_RESERVED_NAME=weu"
yuniql erase -p "c:\play\yuniql-multi-tenant\samples\sqlserver-multi-tenant-sample"  -f -c "Server=localhost,1400;Database=helloyuniql;User Id=SA;Password=P@ssw0rd!" --debug --platform sqlserver  --meta-schema neu  -k "TENANT_RESERVED_NAME=neu"
yuniql erase -p "c:\play\yuniql-multi-tenant\samples\sqlserver-multi-tenant-sample"  -f -c "Server=localhost,1400;Database=helloyuniql;User Id=SA;Password=P@ssw0rd!" --debug --platform sqlserver  --meta-schema apac  -k "TENANT_RESERVED_NAME=apac"

```
